packages/router-core/src/router.ts

```diff
@@ -1173,7 +1173,7 @@ export class RouterCore<
         href: fullPath,
         publicHref: href,
         url: url.href,
-        pathname,
+        pathname: decodePathSegment(pathname),
```

Comment:
Maybe a function rename here, since its no longer `decodePathSegment` (which was using `decodeURIComponent`) rather its more along the lines of `decodePath` (using `decodeURI`).